File: samples/dotnet-isolated-azurefunction/InputBinding/RetrieveSecretLocal.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

namespace dotnet_isolated_azurefunction
{
    using Microsoft.Azure.Functions.Worker;
    using Microsoft.Azure.Functions.Worker.Extensions.Dapr;
    using Microsoft.Extensions.Logging;
    using System.Collections.Generic;

    public static class RetrieveSecretLocal
    {
        /// <summary>
        /// Example to use Dapr Service Invocation Trigger and Dapr Secret input binding to retrieve a saved state from statestore
        /// </summary>
        [Function("RetrieveSecretLocal")]
        public static void Run(
            [DaprServiceInvocationTrigger] object args,
            [DaprSecretInput("localsecretstore", "my-secret", Metadata = "metadata.namespace=default")] IDictionary<string, string> secret,
            FunctionContext functionContext)
        {
            var log = functionContext.GetLogger("RetrieveSecretLocal");
            log.LogInformation("C# function processed a RetrieveSecret request from the Dapr Runtime.");

            // print the fetched secret value
            // this is only for demo purpose
            // please do not log any real secret in your production code        
            foreach (var kvp in secret)
            {
                log.LogInformation("Stored secret: Key = {0}, Value = {1}", kvp.Key, kvp.Value);
            }
        }
    }
}

File: samples/java-azurefunction/src/main/java/com/function/RetrieveSecretLocal.java
package com.function;

import java.util.Map;

import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.dapr.annotation.DaprSecretInput;
import com.microsoft.azure.functions.dapr.annotation.DaprServiceInvocationTrigger;

public class RetrieveSecretLocal {
    @FunctionName("RetrieveSecretLocal")
    public void run(
        @DaprServiceInvocationTrigger(
            methodName = "RetrieveSecretLocal") Object args,
        @DaprSecretInput(
            secretStoreName = "localsecretstore",
            key = "my-secret", 
            metadata = "metadata.namespace=default") 
            Map<String, String> secret,
        final ExecutionContext context) {

        context.getLogger().info("Java function processed a RetrieveSecret request from the Dapr Runtime.");

        // Print the fetched secret value
        // This is only for demo purpose
        // Please do not log any real secret in your production code
        for (Map.Entry<String, String> entry : secret.entrySet()) {
            context.getLogger().info(String.format("Stored secret: Key = %s, Value = %s", entry.getKey(), entry.getValue()));
        }
    }
}
